pkg/nvidia-query/nvml/clock_events.go

```diff
@@ -13,19 +13,20 @@ import (
 	"sigs.k8s.io/yaml"
 
 	apiv1 "github.com/leptonai/gpud/api/v1"
-	"github.com/leptonai/gpud/pkg/log"
 	nvml_lib "github.com/leptonai/gpud/pkg/nvidia-query/nvml/lib"
 )
 
 // Returns true if clock events is supported by all devices.
 // Returns false if any device does not support clock events.
 // ref. undefined symbol: nvmlDeviceGetCurrentClocksEventReasons for older nvidia drivers
 func ClockEventsSupported() (bool, error) {
-	nvmlLib := nvml_lib.NewDefault()
-	if ret := nvmlLib.NVML().Init(); ret != nvml.SUCCESS {
-		return false, fmt.Errorf("failed to initialize NVML: %v", nvml.ErrorString(ret))
+	nvmlLib, err := nvml_lib.New()
```

Comment:
As we talked before, we would like to initialize the NVML once.. For the case, shall we make the function as a method of InstanceV2? or provide a free function required passing the InstanceV2 into it?

Reply:
Yeah I am also not a fan of current interface. Let us clean up in the following PR https://github.com/leptonai/gpud/issues/511